File: app/src/main/java/org/mifos/mobile/ui/activities/SplashActivity.java
package org.mifos.mobile.ui.activities;

/*
 * Created by saksham on 01/June/2018
 */

import android.content.Intent;
import android.os.Bundle;

import com.mifos.mobile.passcode.utils.PasscodePreferencesHelper;

import org.mifos.mobile.api.local.PreferencesHelper;
import org.mifos.mobile.ui.activities.base.BaseActivity;
import org.mifos.mobile.utils.Constants;

public class SplashActivity extends BaseActivity {

    PasscodePreferencesHelper passcodePreferencesHelper;
    PreferencesHelper prefs;
    Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getActivityComponent().inject(this);

        passcodePreferencesHelper = new PasscodePreferencesHelper(this);
        prefs = new PreferencesHelper(this);
        if (!passcodePreferencesHelper.getPassCode().isEmpty()) {
            intent = new Intent(this, PassCodeActivity.class);
            intent.putExtra(Constants.INTIAL_LOGIN, true);
        } else if (passcodePreferencesHelper.getPassCode().isEmpty() &&
                prefs.getBoolean("check_first_time", true)) {
            intent = new Intent(this, IntroductionActivity.class);
        } else {
            intent = new Intent(this, LoginActivity.class);
        }
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/org/mifos/mobile/ui/activities/SplashActivity.kt
package org.mifos.mobile.ui.activities

import android.content.Intent
import android.os.Bundle
import com.mifos.mobile.passcode.utils.PasscodePreferencesHelper
import org.mifos.mobile.ui.activities.base.BaseActivity
import org.mifos.mobile.ui.login.LoginActivity
import org.mifos.mobile.utils.Constants

/*
* Created by saksham on 01/June/2018
*/
class SplashActivity : BaseActivity() {

    private var passcodePreferencesHelper: PasscodePreferencesHelper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)


        passcodePreferencesHelper = PasscodePreferencesHelper(this)
        val intent: Intent = if (passcodePreferencesHelper?.passCode?.isNotEmpty() == true) {
            Intent(this, PassCodeActivity::class.java).apply {
                putExtra(Constants.INTIAL_LOGIN, true)
            }
        } else {
            Intent(this, LoginActivity::class.java)
        }

        startActivity(intent)
        finish()
    }




    }

